app/src/components/EnhancedComponent.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+import {ReactNode} from "react";
+
+abstract class EnhancedComponent<P extends IEnhancedComponentProps = IEnhancedComponentProps, S extends IEnhancedComponentState = IEnhancedComponentState> extends React.PureComponent<P, S> {
```

Comment:
Typescript makes this look more complicated than necessary, but I guess that's an ok tradeoff

Reply:
it's just these 2 files that'll have it look like this, others will be shorter